pkg/jobs/jobs_test.go

```diff
@@ -0,0 +1,271 @@
+package jobs
+
+import (
+	"github.com/grafana/xk6-kubernetes/internal/testutils"
+	k8sTypes "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/fake"
+	"strings"
+	"testing"
+)
+
+var (
+	testName      = "job-test"
+	testNamespace = "ns-test"
+)
+
+func TestJobs_Apply(t *testing.T) {
+	t.Parallel()
+	fixture := New(fake.NewSimpleClientset(
+		testutils.NewJob("existing", testNamespace),
+	), metav1.ListOptions{}, nil)
+
+	testCases := []struct {
+		testID        string
+		yaml          string
+		namespace     string
+		expectedError string
+	}{
+		{
+			testID: "apply with invalid yaml",
+			yaml: `
+this is not yaml
+`,
+			namespace:     testNamespace,
+			expectedError: "json parse error",
+		},
+		{
+			testID: "apply with incorrect resource yaml",
```

Comment:
@javaducky What is the purpose of these tests with invalid yaml? It seems to me you are testing there the fixture, not the jobs package.